src/CarPlay/CPCompat.cs

```diff
@@ -0,0 +1,35 @@
+//
+// CPCompat.cs
+//
+// Authors:
+//	Alex Soto <alexsoto@microsoft.com>
+//
+// Copyright (c) Microsoft Corporation.
+//
+
+using System;
+using Foundation;
+using ObjCRuntime;
+using System.ComponentModel;
+
+#if !XAMCORE_4_0
+namespace CarPlay {
+	[Register (SkipRegistration = true)]
+	[Introduced (PlatformName.iOS, 14, 0)]
+	[Unavailable (PlatformName.iOS, PlatformArchitecture.All)]
+	[Obsolete ("'CPEntity' has been removed from iOS.")]
+	[EditorBrowsable (EditorBrowsableState.Never)]
+	public class CPEntity : NSObject, INSSecureCoding {
+
+		public CPEntity () => throw new NotImplementedException ();
```

Comment:
I think we usually throw `NotSupportedException` instead of `NotImplementedException`.